custom_enumerable_methods.rb

```diff
@@ -0,0 +1,113 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    for element in self # rubocop:disable Style/For
+      yield(element)
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    my_self = self
+    i = 0
+    element = 0
+    my_self.my_each do
+      yield(element, i)
+      i += 1
+    end
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    result = []
+    my_self = self
+    my_self.my_each { |i| result.push(i) if yield(i) }
+    result
+  end
+
+  def my_all?(*arg)
+    result = true
+    if !arg[0].nil?
+      my_each { |i| result = false unless arg[0] == i }
+    elsif !block_given?
+      my_each { |i| result = false unless i }
+    else
+      my_each { |i| result = false unless yield(i) }
+    end
+    result
+  end
```

Comment:
```rb
2.6.5 :033 > array
 => [3, 4, 2, 6, 8, 4, 6, 5, 1, 2, 4, 1, 5, 6, 7, 8, 1, 7, 3, 0, 8, 0, 3, 6, 0, 1, 0, 6, 2, 6, 3, 7, 3, 2, 3, 5, 1, 1, 7, 8, 2, 3, 4, 8, 1, 8, 0, 3, 4, 5, 4, 3, 3, 7, 1, 8, 4, 7, 8, 2, 6, 8, 0, 6, 0, 5, 1, 2, 5, 2, 2, 4, 0, 2, 1, 8, 1, 3, 3, 8, 2, 8, 1, 6, 1, 0, 1, 2, 5, 0, 5, 8, 1, 3, 2, 7, 4, 5, 2, 4] 
2.6.5 :034 > array.my_all?(Integer)
 => false 
2.6.5 :035 > array.all?(Integer)
 => true 
2.6.5 :036 >
```

Reply:
```rb
2.6.5 :038 > words = %w[dog door rod blade]
2.6.5 :039 > words.my_all?(/d/)
 => false 
2.6.5 :040 > words.all?(/d/)
 => true 
2.6.5 :041 >
```

Reply:
# Thank You! :bouquet: 
## Above issue is fixed now! :heavy_check_mark: 

## I helped us to learn more
### Thank You so much! :bouquet: